jupyterhub/templates/hub/configmap.yaml

```diff
@@ -133,6 +133,9 @@ data:
     {{- .Values.singleuser.cloudMetadata | toYaml | trimSuffix "\n" | nindent 4 }}
   singleuser.start-timeout: {{ .Values.singleuser.startTimeout | quote }}
   singleuser.image-pull-policy: {{ .Values.singleuser.image.pullPolicy | quote }}
+  {{- if .Values.singleuser.image.pullSecret }}
+  singleuser.image-pull-secret: {{ .Values.singleuser.image.pullSecret | quote }}
```

Comment:
The value `.Values.singleuser.image.pullSecret` is not just a string so it can't be added like this, but you must instead use some `toYaml` trick. Something like this:

```yaml
   {{- if .Values.auth.github.org_whitelist }}
   auth.github.org_whitelist: |
     {{- .Values.auth.github.org_whitelist | toYaml | trimSuffix "\n" | nindent 4 }}
   {{- end }}
```

Reply:
sorry this was supposed to be  {{- if .Values.singleuser.image.pullSecret.name }}

Reply:
Ah! I looked at the source code of Kubespawner and [Kubernets API](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#podspec-v1-core) and learned some things!

I learned that KubeSpawner's setting `image_pull_secrets` is not named properly, it should really be named `image_pull_secret_name`. I suggested below we could name the secret `singleuser-image-credentials`, this is what we need to assign to `singleuser.image-pull-secret`, as this is used to set KubeSpawners setting in the `jupyterhub_config.py` file.

Reply:
yeah i dug all yesterday. quite a bit of layers. take a look at what i have now and let me know if that makes sense.